src/shared.c

```diff
@@ -191,7 +191,7 @@ stl_write_vrml(stl_file *stl, char *file) {
                 malloc(81 + strlen(file)); /* Allow 80 chars+file size for message */
     sprintf(error_msg, "stl_write_ascii: Couldn't open %s for writing",
             file);
-    perror(error_msg);
+    stl_write_error_message(stl, error_msg);
```

Comment:
The gotcha here is that for IO operations, perror provided additional context (permission error, file not found error, etc.) so for such cases, we should probably combine that info with our own.

Reply:
So what do you suggest? In the issue thread in the comment you discouraged usage of the perror, so maybe we can write our error message to the stderr and then use `strerror(errno)` to get the error context string you mentioned.

Reply:
I guess we need to combine strerror with our error_msg. Something like (psoudo code):

```
errno = errno();
if errno != 0 {
    stl_write_error_message(stl, strerror(errno) + error_msg);
} else {
    stl_write_error_message(stl, error_msg);
}
```

Reply:
Alright, I added the concatenation code directly to the `stl_write_error_message` function to not repeat errno checks unnecessarily.